src/main/java/de/unistuttgart/iste/gits/course_service/controller/CourseController.java

```diff
@@ -44,7 +44,18 @@ public List<Course> coursesByIds(@Argument(name = "ids") final List<UUID> ids) {
     @MutationMapping
     public Course createCourse(@Argument(name = "input") final CreateCourseInput input,
                                @ContextValue final LoggedInUser currentUser) {
-        return courseService.createCourse(input, currentUser.getId());
+        final Course course = courseService.createCourse(input, currentUser.getId());
+
+        // update user course memberships in context with the newly created course (the creator of the course
+        // always gets admin permissions)
+        currentUser.getCourseMemberships().add(new LoggedInUser.CourseMembership(
```

Comment:
why?

Basically the same code already exists in the CourseService (line 47), and even if it wouldn't it should be added there and not in the controller

Reply:
the problem is the `chapters @SchemaMapping`, because it requires the user to be a member of the course. Membership is determined by the courseMemberships in the GraphQL context. But this obviously isn't refreshed when the course is created.

The other option would be to re-fetch course memberships for each invocation of the chapters schema mapping, but that would be worse from a performance standpoint.

Reply:
Can you quickly explain why you add the new course to the current User?

Is it so if you put multiple queries in one request, it still works?